Dockerfile

```diff
@@ -51,14 +52,15 @@ RUN git clone --recurse-submodules -b ${CUDF_BRANCH} ${CUDF_REPO} /cudf
 # libcudf build/install
 ENV CC=/usr/bin/gcc-${CC}
 ENV CXX=/usr/bin/g++-${CXX}
-ARG HASH_JOIN=ON
 RUN source activate cudf && \
     mkdir -p /cudf/cpp/build && \
     cd /cudf/cpp/build && \
-    cmake .. -DHASH_JOIN=${HASH_JOIN} -DCMAKE_INSTALL_PREFIX=${CONDA_PREFIX} && \
-    make -j2 install && \
+    cmake .. -DCMAKE_INSTALL_PREFIX=${CONDA_PREFIX} && \
+    make -j install && \
     make python_cffi && \
-    make install_python
+    make install_python && \
+    cd python && \
+    python setup.py install
```

Comment:
`python setup.py install` is run by `make install_python`, so this change is redundant.

Reply:
And as mentioned in another comment, you are missing the steps to build the Cython bindings. 

```
cd /cudf/python && python setup.py build_ext --inplace && python setup.py install

Reply:
[Per Python docs](https://docs.python.org/3/install/), `python setup.py install` actually performs both the build and install step in one command.

`python setup.py build_ext --inplace` builds the bindings and only places them in the source folder. It does not install them.

I'm not sure if `make install_python` has an issue, but if you do only that, and don't do `python setup.py install` as in this PR, you get the [librmm import issue](https://github.com/rapidsai/cudf/issues/431).

Output when only running `make install_python` and `python setup.py build_ext --inplace`:
```
root@af88c1745732:/# source activate cudf
(cudf) root@af88c1745732:~# python
Python 3.5.6 |Anaconda, Inc.| (default, Aug 26 2018, 21:41:56) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import cudf
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/conda/envs/cudf/lib/python3.5/site-packages/cudf-0.3.0+138.gdae6026-py3.5-linux-x86_64.egg/cudf/__init__.py", line 2, in <module>
    from cudf import dataframe             # noqa: F401
  File "/conda/envs/cudf/lib/python3.5/site-packages/cudf-0.3.0+138.gdae6026-py3.5-linux-x86_64.egg/cudf/dataframe/__init__.py", line 1, in <module>
    from cudf.dataframe import (buffer, dataframe, series,  # noqa: F401
  File "/conda/envs/cudf/lib/python3.5/site-packages/cudf-0.3.0+138.gdae6026-py3.5-linux-x86_64.egg/cudf/dataframe/buffer.py", line 3, in <module>
    from librmm_cffi import librmm as rmm
ImportError: cannot import name 'librmm'
```

Reply:
Interesting. Here's the relevant lines from libcudf CMakeLists.txt:
https://github.com/rapidsai/cudf/blob/dae6026475aa19947fcf46c792cd11a18a32c43b/cpp/CMakeLists.txt#L214-L219

You can see that it definitely runs `python setup.py install` in the cpp/python directory. Perhaps there is a dependency issue and it thinks it doesn't need to be run? Can you try modifying `make install_python` line to this and share the output?

`make install_python VERBOSE=1`

Reply:
```
[ 93%] Built target cudf
make -f CMakeFiles/install_python.dir/build.make CMakeFiles/install_python.dir/depend
make[3]: Entering directory '/cudf/cpp/build'
cd /cudf/cpp/build && /conda/envs/cudf/bin/cmake -E cmake_depends "Unix Makefiles" /cudf/cpp /cudf/cpp /cudf/cpp/build /cudf/cpp/build /cudf/
cpp/build/CMakeFiles/install_python.dir/DependInfo.cmake --color=
Dependee "/cudf/cpp/build/CMakeFiles/install_python.dir/DependInfo.cmake" is newer than depender "/cudf/cpp/build/CMakeFiles/install_python.d
ir/depend.internal".
Dependee "/cudf/cpp/build/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/cudf/cpp/build/CMakeFiles/install_python.dir/d
epend.internal".
Scanning dependencies of target install_python
make[3]: Leaving directory '/cudf/cpp/build'
make -f CMakeFiles/install_python.dir/build.make CMakeFiles/install_python.dir/build
make[3]: Entering directory '/cudf/cpp/build'
[ 96%] Generating PYTHON_CFFI
/conda/envs/cudf/bin/cmake -E copy_directory /cudf/cpp/build/../python /cudf/cpp/build/python
/conda/envs/cudf/bin/cmake -E chdir /cudf/cpp/build/python python setup.py install --single-version-externally-managed --record=record.txt
running install
running build
running build_py
file libgdf_cffi/libgdf_cffi.py (for module libgdf_cffi.libgdf_cffi) not found
copying libgdf_cffi/__init__.py -> build/lib/libgdf_cffi
copying libgdf_cffi/wrapper.py -> build/lib/libgdf_cffi
copying libgdf_cffi/libgdf_build.py -> build/lib/libgdf_cffi
file libgdf_cffi/libgdf_cffi.py (for module libgdf_cffi.libgdf_cffi) not found
generating cffi module 'build/lib/libgdf_cffi/libgdf_cffi.py'
already up-to-date
running install_lib
copying build/lib/libgdf_cffi/wrapper.py -> /conda/envs/cudf/lib/python3.5/site-packages/libgdf_cffi
copying build/lib/libgdf_cffi/libgdf_build.py -> /conda/envs/cudf/lib/python3.5/site-packages/libgdf_cffi
copying build/lib/libgdf_cffi/__init__.py -> /conda/envs/cudf/lib/python3.5/site-packages/libgdf_cffi
running install_egg_info
running egg_info
writing top-level names to libgdf_cffi.egg-info/top_level.txt
writing requirements to libgdf_cffi.egg-info/requires.txt
writing libgdf_cffi.egg-info/PKG-INFO
writing dependency_links to libgdf_cffi.egg-info/dependency_links.txt
file libgdf_cffi/libgdf_cffi.py (for module libgdf_cffi.libgdf_cffi) not found
reading manifest file 'libgdf_cffi.egg-info/SOURCES.txt'
writing manifest file 'libgdf_cffi.egg-info/SOURCES.txt'
removing '/conda/envs/cudf/lib/python3.5/site-packages/libgdf_cffi-0.3.0-py3.5.egg-info' (and everything under it)
Copying libgdf_cffi.egg-info to /conda/envs/cudf/lib/python3.5/site-packages/libgdf_cffi-0.3.0-py3.5.egg-info
running install_scripts                                                                                                            [448/1970]
file libgdf_cffi/libgdf_cffi.py (for module libgdf_cffi.libgdf_cffi) not found
writing list of installed files to 'record.txt'
running install
running build
running build_py
file librmm_cffi/librmm_cffi.py (for module librmm_cffi.librmm_cffi) not found
copying librmm_cffi/__init__.py -> build/lib/librmm_cffi
copying librmm_cffi/librmm_build.py -> build/lib/librmm_cffi
copying librmm_cffi/wrapper.py -> build/lib/librmm_cffi
copying librmm_cffi/librmm_config.py -> build/lib/librmm_cffi
file librmm_cffi/librmm_cffi.py (for module librmm_cffi.librmm_cffi) not found
generating cffi module 'build/lib/librmm_cffi/librmm_cffi.py'
already up-to-date
running install_lib
copying build/lib/librmm_cffi/librmm_build.py -> /conda/envs/cudf/lib/python3.5/site-packages/librmm_cffi
copying build/lib/librmm_cffi/wrapper.py -> /conda/envs/cudf/lib/python3.5/site-packages/librmm_cffi
copying build/lib/librmm_cffi/librmm_config.py -> /conda/envs/cudf/lib/python3.5/site-packages/librmm_cffi
copying build/lib/librmm_cffi/__init__.py -> /conda/envs/cudf/lib/python3.5/site-packages/librmm_cffi
running install_egg_info
running egg_info
writing top-level names to librmm_cffi.egg-info/top_level.txt
writing requirements to librmm_cffi.egg-info/requires.txt
writing librmm_cffi.egg-info/PKG-INFO
writing dependency_links to librmm_cffi.egg-info/dependency_links.txt
file librmm_cffi/librmm_cffi.py (for module librmm_cffi.librmm_cffi) not found
reading manifest file 'librmm_cffi.egg-info/SOURCES.txt'
writing manifest file 'librmm_cffi.egg-info/SOURCES.txt'
removing '/conda/envs/cudf/lib/python3.5/site-packages/librmm_cffi-0.3.0-py3.5.egg-info' (and everything under it)
Copying librmm_cffi.egg-info to /conda/envs/cudf/lib/python3.5/site-packages/librmm_cffi-0.3.0-py3.5.egg-info
running install_scripts
file librmm_cffi/librmm_cffi.py (for module librmm_cffi.librmm_cffi) not found
writing list of installed files to 'record.txt'
[100%] Generating INSTALL_PYTHON_CFFI
/conda/envs/cudf/bin/cmake -E chdir /cudf/cpp/build/python python setup.py install --single-version-externally-managed --record=record.txt
running install
running build
running build_py
file libgdf_cffi/libgdf_cffi.py (for module libgdf_cffi.libgdf_cffi) not found
file libgdf_cffi/libgdf_cffi.py (for module libgdf_cffi.libgdf_cffi) not found
generating cffi module 'build/lib/libgdf_cffi/libgdf_cffi.py'
already up-to-date
running install_lib
running install_egg_info                                                                                                           [405/1970]
running egg_info
writing requirements to libgdf_cffi.egg-info/requires.txt
writing dependency_links to libgdf_cffi.egg-info/dependency_links.txt
writing top-level names to libgdf_cffi.egg-info/top_level.txt
writing libgdf_cffi.egg-info/PKG-INFO
file libgdf_cffi/libgdf_cffi.py (for module libgdf_cffi.libgdf_cffi) not found
reading manifest file 'libgdf_cffi.egg-info/SOURCES.txt'
writing manifest file 'libgdf_cffi.egg-info/SOURCES.txt'
removing '/conda/envs/cudf/lib/python3.5/site-packages/libgdf_cffi-0.3.0-py3.5.egg-info' (and everything under it)
Copying libgdf_cffi.egg-info to /conda/envs/cudf/lib/python3.5/site-packages/libgdf_cffi-0.3.0-py3.5.egg-info
running install_scripts
file libgdf_cffi/libgdf_cffi.py (for module libgdf_cffi.libgdf_cffi) not found
writing list of installed files to 'record.txt'
running install
running build
running build_py
file librmm_cffi/librmm_cffi.py (for module librmm_cffi.librmm_cffi) not found
file librmm_cffi/librmm_cffi.py (for module librmm_cffi.librmm_cffi) not found
generating cffi module 'build/lib/librmm_cffi/librmm_cffi.py'
already up-to-date
running install_lib
running install_egg_info
running egg_info
writing requirements to librmm_cffi.egg-info/requires.txt
writing dependency_links to librmm_cffi.egg-info/dependency_links.txt
writing top-level names to librmm_cffi.egg-info/top_level.txt
writing librmm_cffi.egg-info/PKG-INFO
file librmm_cffi/librmm_cffi.py (for module librmm_cffi.librmm_cffi) not found
reading manifest file 'librmm_cffi.egg-info/SOURCES.txt'
writing manifest file 'librmm_cffi.egg-info/SOURCES.txt'
removing '/conda/envs/cudf/lib/python3.5/site-packages/librmm_cffi-0.3.0-py3.5.egg-info' (and everything under it)
Copying librmm_cffi.egg-info to /conda/envs/cudf/lib/python3.5/site-packages/librmm_cffi-0.3.0-py3.5.egg-info
running install_scripts
file librmm_cffi/librmm_cffi.py (for module librmm_cffi.librmm_cffi) not found
writing list of installed files to 'record.txt'
make[3]: Leaving directory '/cudf/cpp/build'
[100%] Built target install_python
make[2]: Leaving directory '/cudf/cpp/build'
/conda/envs/cudf/bin/cmake -E cmake_progress_start /cudf/cpp/build/CMakeFiles 0
make[1]: Leaving directory '/cudf/cpp/build'
Removing intermediate container 2866f3a29380
 ---> 4cfaa3eaaa17
Step 27/28 : ADD python /cudf/python
 ---> 622a48fc4878
Step 28/28 : RUN source activate cudf &&     cd /cudf/python &&     python setup.py install
 ---> Running in 2b6ff8c3f451
...
```

Reply:
Right, so it's already running `python setup.py install` for libcudf. You don't need to call it explicitly.

Reply:
I tried again, without the `--single-version-externally-managed`, and it looks like it worked. Updating this PR now.